### PR TITLE
fix mitering bug with ground polylines over long distances

### LINKED ISSUE #7855: GroundPolylineGeometry missing/occluded segment
GroundPolylinePrimitive/Geometry with specific points produces a missing or occluded segment (with or without terrain; also with `GEODESIC` or `RHUMB` arc types). In the following example, changing the second vertex latitude to `48.499` fixes the issue.

Sandcastle example: [Sandcastle example](https://cesiumjs.org/Cesium/Build/Apps/Sandcastle/#c=dZPbjtowEIZfxeJmE4kack4oi4pYRFdCYqVdtapKL0wygFVjI9sBZSvevTaH3XCaiyTjme/3eCbeEIk2FLYg0SPisEUDULRc4R/7Nech37sDwTWhHOSD+3XKp/yYNJKi5MWLYBUzwRdJV1TTDWDKzZsw+g5vIKUBvwNdLLVyXKyXwJ15yXNNBXdc9G/KkbFDCVjlwAEvDrInOYVJUTi12u5s6xy1rBFV8XwpBRel6qA5YQqateh6DUQSnkOnfuaT4kAwIfsfOY5bQxcgVqBl9cyVtkF1pjC6iNZLqtPn0NlxThKXqLW1UNT2zWx6ZAdEavNFeIDnUqyeYCEBVN80vXJ+XwtY++K1I5z4STvMvCyIgiSOmij0ceT5UZRkaZDGSdq8y7ZxkvjtwPJeZmQMm+IwS7M0NnKpn4TePTgLceQHcZbEaRh6JreJohD7npfFSeIlcRiEWXzN/nFvCBKZv1Vr+GhE/+Dj0XDyNHx9HtxAtrTQyw7yzyO7S3WitaSzUtvR3phBbn+Oz/5b73Lq/ZPCfib7FKeej38Nx+PJT/eikE93dwztXHvddvbRaDa6SlcMeofQN7paC6lRKZmDcUvDas2Iqbk1K/O/oHGulMW6rRPULegG0eJx2ri40tMGyhlRykTmJWOv5tZOG71uy+SfYUyQgvLFZAOSkcqmLL3e+LCIMe62jHtNaSHYjMia4n8)

Browser: Chrome 74.0.3729.131 and Firefox 66.0.5

Operating System: OS X 10.14.4

Screen Shot 2019-05-20 at 4 05 50 PM

Edit:
This is easily reproducible with lines which are already interpolated along geodesics, but not with the same granularity as Cesium. The items producing this issue in our application are geodesic lines between points with a granularity of 100km.

### PR DESCRIPTION
**Pull request by likangning93**
Fixes #7855.

`GroundPolylineGeometry` miters between line segments, but only if their endpoints aren't coplanar.
This used to be a literal coplanar check, but this would fail sometimes for very long-distance line segments due to the tiny epsilon used for coplanarity. The computed miter vector would then also have problems due to the curvature of the Earth, since it was a naive average of the directions from point-to-point.

This PR moves the check for whether miters are needed or not to a check using vectors pointing towards the segments' end points but tangent to the surface of the ellipsoid.

The spec for miter breaking was also incorrect, the first test case wasn't even encountering a miter break.

### CODE REVIEW

**Comment by cesium-concierge** [comment]
Thanks for the pull request @likangning93!
* :heavy_check_mark: Signed CLA found.

Reviewers, don't forget to make sure that:

- [ ] Cesium Viewer works.
- [ ] Works in 2D/CV.
- [ ] Works (or fails gracefully) in IE11.

**Comment by wallw-bits** [comment]
That fixes our issue. Thanks!

File: Source/Core/GroundPolylineGeometry.js [hunk 1]
```diff
@@ -353,41 +353,42 @@ define([
...
+    }
+
     var toPreviousScratch = new Cartesian3();
     var toNextScratch = new Cartesian3();
     var forwardScratch = new Cartesian3();
-    var coplanarNormalScratch = new Cartesian3();
-    var coplanarPlaneScratch = new Plane(Cartesian3.UNIT_X, 0.0);
     var vertexUpScratch = new Cartesian3();
     var cosine90 = 0.0;
+    var cosine180 = -1.0;
```
Comment (likangning93) [inline Source/Core/GroundPolylineGeometry.js]: @lilleyse for context, `computeVertexMiterNormal` is meant to compute a normal pointing to the "right" side of a join between two line segments defined by 3 points, eventually this is used to compute normals on the start and end faces of the volumes around polyline segments.

**Comment by lilleyse** [comment]
Looks good!

**Pull request merged**